cw_bitcoin/lib/electrum_wallet_addresses.dart

```diff
@@ -132,19 +145,24 @@ abstract class ElectrumWalletAddressesBase extends WalletAddresses with Store {
     return address;
   }
 
+  @action
   BitcoinAddressRecord generateNewAddress(
-      {bitcoin.HDWallet? hd, bool isHidden = false}) {
-    currentReceiveAddressIndex += 1;
+      {bitcoin.HDWallet? hd, bool isHidden = false, String? label}) {
     // FIX-ME: Check logic for whichi HD should be used here  ???
     final address = BitcoinAddressRecord(
-        getAddress(index: currentReceiveAddressIndex, hd: hd ?? sideHd),
+        getAddress(
+            index: currentReceiveAddressIndex, hd: hd ?? sideHd, addressType: addressPageType),
         index: currentReceiveAddressIndex,
         isHidden: isHidden);
     addresses.add(address);
     return address;
+
+    currentReceiveAddressIndex += 1;
```

Comment:
shouldn't this be before the `getAddress` call?
like this it will generate the same address for the current users (at least for the first time after they update)